packages/analytics/src/integrations/Omnitracking/__tests__/omnitracking-helper.test.ts

```diff
@@ -237,3 +243,63 @@ describe('getGenderValueFromProperties', () => {
     ).toBe('W,M');
   });
 });
+
+describe('getRecommendationsTrackingData', () => {
+  const mockedRecommendationsValues = {
+    list: 'list123',
+    listId: '09a35590-bb62-4027-a630-5da04ec64fb5',
+  };
+
+  const mockedReturnedRecommendationsValues = {
+    moduleTitle: JSON.stringify([mockedRecommendationsValues.list]),
+    moduleId: JSON.stringify([mockedRecommendationsValues.listId]),
+  };
+
+  it('should return undefined in case its non recommendations product-details event', () => {
+    expect(
+      getRecommendationsTrackingData({
+        event: PageType.ProductDetails,
+        properties: {
+          from: 'abc',
+        } as Record<string, unknown>,
+      } as EventData<TrackTypesValues>),
+    ).toBeUndefined();
+  });
+
+  it('should return empty object if the event is sent without the recommendations parameters filled', () => {
+    expect(
+      getRecommendationsTrackingData({
+        event: 'abc',
+        properties: {
+          from: FromParameterType.Recommendations,
+        } as Record<string, unknown>,
+      } as EventData<TrackTypesValues>),
+    ).toEqual({});
+  });
+
+  it('should return the json stringified values correctly', () => {
+    expect(
+      getRecommendationsTrackingData({
+        event: 'abc',
+        properties: {
+          from: FromParameterType.Recommendations,
+          ...mockedRecommendationsValues,
+        } as Record<string, unknown>,
+      } as EventData<TrackTypesValues>),
+    ).toEqual({ ...mockedReturnedRecommendationsValues });
```

Comment:
Spread not needed.

`expect(
       getRecommendationsTrackingData({
         event: 'abc',
         properties: {
           from: FromParameterType.Recommendations,
           ...mockedRecommendationsValues,
         } as Record<string, unknown>,
       } as EventData<TrackTypesValues>),
     ).toEqual(mockedReturnedRecommendationsValues);`

Reply:
Fixed 👍🏼